qiita_ware/ebi.py

```diff
@@ -872,50 +853,68 @@ def generate_demultiplexed_fastq(self, rewrite_fastq=False, mtime=None):
 
         Notes
         -----
-        - As a performace feature, this method will check if self.ebi_dir
+        - As a performace feature, this method will check if self.full_ebi_dir
         already exists and, if it does, the script will assume that in a
         previous execution this step was performed correctly and will simply
-        read the file names from self.ebi_dir
+        read the file names from self.full_ebi_dir
         - When the object is created (init), samples, samples_prep and
         sample_demux_fps hold values for all available samples in the database.
         Here some of those values will be deleted (del's, within the loops) for
         those cases where the fastq.gz files weren't written or exist. This is
         an indication that they had no sequences and this kind of files are not
         accepted in EBI
+
+        Raises
+        ------
+        EBISubmissionError
+            - The demux file couldn't be read
+            - All samples are removed
         """
         ppd = self.preprocessed_data
 
-        if not isdir(self.ebi_dir) or rewrite_fastq:
-            makedirs(self.ebi_dir)
+        dir_not_exists = not isdir(self.full_ebi_dir)
+        if dir_not_exists or rewrite_fastq:
+            if dir_not_exists:
+                makedirs(self.full_ebi_dir)
+                filetype_id = convert_to_id('directory', 'filepath_type')
+                insert_filepaths([(self.full_ebi_dir, filetype_id)], -1,
+                                 'preprocessed_data', 'filepath',
+                                 move_files=False)
 
             demux = [path for _, path, ftype in ppd.get_filepaths()
                      if ftype == 'preprocessed_demux'][0]
 
             demux_samples = set()
-            with open_file(demux) as demux_fh:
-                for s, i in to_per_sample_ascii(demux_fh,
-                                                self.prep_template.keys()):
-                    sample_fp = self.sample_demux_fps[s]
-                    wrote_sequences = False
-                    with GzipFile(sample_fp, mode='w', mtime=mtime) as fh:
-                        for record in i:
-                            fh.write(record)
-                            wrote_sequences = True
-
-                    if wrote_sequences:
-                        demux_samples.add(s)
-                    else:
-                        del(self.samples[s])
-                        del(self.samples_prep[s])
-                        del(self.sample_demux_fps[s])
-                        remove(sample_fp)
-
+            try:
```

Comment:
Is the reasoning for this block the fact that absolutely any of these statements can fail and that they have been known to fail before? If at all possible, I would try to change this to catch specific exceptions and the `Exception` class, else, it we would need to include the full traceback. I'm trying to think what to do in a case where we have to debug this because it failed at some point.